test/Chainweb/Test/Pact/Utils.hs

```diff
@@ -514,3 +521,33 @@ withBlockHeaderDb iordb b = withResource start stop
 
 withTemporaryDir :: (IO FilePath -> TestTree) -> TestTree
 withTemporaryDir = withResource (fst <$> newTempDir) removeDirectoryRecursive
+
+withPact
+    :: ChainwebVersion
+    -> LogLevel
+    -> IO (PayloadDb HashMapCas)
+    -> IO BlockHeaderDb
+    -> MemPoolAccess
+    -> IO FilePath
+    -> (IO (TQueue RequestMsg) -> TestTree)
+    -> TestTree
+withPact version logLevel iopdb iobhdb mempool iodir f =
+    withResource startPact stopPact $ f . fmap snd
+  where
+    startPact = do
+        mv <- newEmptyMVar
+        reqQ <- atomically newTQueue
+        pdb <- iopdb
+        bhdb <- iobhdb
+        dir <- iodir
+        a <- async $ initPactService version cid logger reqQ mempool mv
```

Comment:
please use the `withLink` utility for all linked async work

Reply:
Oh.. of course.